pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -24,6 +23,13 @@ import (
 	"github.com/openshift/origin/pkg/client"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 	"github.com/openshift/origin/pkg/image/importer"
+	quotautil "github.com/openshift/origin/pkg/quota/util"
+)
+
+const (
+	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying external url of internal
+	// docker registry. All references to pushed images will be prefixed with its value.
+	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
```

Comment:
Be careful with that, what if we allow accessing registry by name (planned) or IP (currently availabel)? Both should work.